internal/querynode/query_node.go

```diff
@@ -266,6 +266,14 @@ func (node *QueryNode) Init() error {
 			return
 		}
 
+		node.chunkManager, initError = storage.NewMinioChunkManager(node.queryNodeLoopCtx,
```

Comment:
if initError then be need to break initOnce, because other people may reuse initError and overwrite error message

Reply:
It is better to have a set of timeout control mechanism to connect to minio or etcd. Our system does not have a unified set of rules for the timeout. In most cases, the timeout mechanism is implemented by retry. Now it is implemented in NewMinioChunkManager and retries 300 times.
I haven't thought about metrics related things. I have seen how to get minio metrics on minio's official website. Maybe we can add some additional metrics we want on the basis of official metrics.